### ISSUE TITLE
Incorrect suggestions with OverloadedLists

### ISSUE THREAD

**Issue by expipiplus1** [ndmitchell/hlint#114]
``` Haskell
{-# LANGUAGE OverloadedLists #-}
module Baz where

import Data.Foldable

foo :: Foldable t => t a -> t a
foo = id

bar = foo (True:[])
```

HLint suggests: 

```
Warning: Use list literal
Found:
  True : []
Why not:
  [True]
```

This will prevent the program from typechecking due to the ambiguity introduced by OverloadedLists

**Comment by ndmitchell** [comment]
Thanks for the report. I guess removing (:) takes you to the generalisation. I guess I should check for overloaded lists and not suggest the replacement.
